### DISCUSSION TITLE
Contract verification on zkSync Sepolia Testnet fails

### DISCUSSION THREAD

**Discussion by SanguineTitan57** [Cyfrin/foundry-full-course-cu#2749] [Q&A] [answered]
I have been trying to deploy and verify. Deployment is successful, but verification fails each time.
I have even changed the solidity compiler to match that of zkSync's compiler.
Nothing works

I get this error:
``Error in plugin @matterlabs/hardhat-zksync-verify: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.``

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.`
        
Each of the highlighted issues are resolved/never existed.
I also did get an error on zkSync's plugin being outdated? Could that be the problem.

**Comment by SanguineTitan57** [comment, marked as answer]
![Screenshot 2024-10-25 163448](https://github.com/user-attachments/assets/354bf801-4af9-4fca-81ad-0e1ba9092067)

The issue was that I hadn't put my SimpleStorage.sol file within a contracts directory.
Hope this helps someone!

**Reply by EngrPips** [reply]
Awesome, I am glad you figured this out.
